tiny_dnn/core/kernels/tiny_quantized_conv2d_kernel.h

```diff
@@ -135,25 +97,40 @@ inline void tiny_quantized_conv2d_kernel(const conv_params& params,
                 }
             }
         }
-        if (params.has_bias) {
-            int32_t * pa_quantized  = &a_quantized[params.out.get_index(0, 0, o)];
-            int32_t * paa_quantized = pa_quantized + params.out.width_ * params.out.height_;
-            std::for_each(pa_quantized, paa_quantized, [&](int32_t& f) {
-                f += (bias_quantized[o] - zero_in_total_space);
-            });
-        }
     });
 
-    float_t min_output_requantized;
-    float_t max_output_requantized;
+    vec_t range_output_requantized(2, 0);
     std::vector<uint8_t> a_requantized(a_quantized.size(), static_cast<uint8_t>(0));
 
-    // Requantize from 32bits to 8 bits for next layer
-    quantize_down_and_shrink_range<int32_t, uint8_t>(a_quantized, min_output_value, max_output_value,
-    &min_output_requantized, &max_output_requantized, &a_requantized);
+    // Requantize from 32bits to 8 bits
+    quantize_down_and_shrink_range<int32_t, uint8_t>(a_quantized,
+        range_output[0], range_output[1],
+        &(range_output_requantized[0]), &(range_output_requantized[1]),
+        &a_requantized);
 
-    // dequantize to flaot, this could be removed within concatenated quantized network
-    a = quantized_tensor_to_float<uint8_t>(a_requantized, min_output_requantized, max_output_requantized);
+    // Adding bias if it is needed
+    vec_t range_b(2, 0);
+    std::vector<uint8_t> b_quantized;
+    if (params.has_bias) {
+      // Bias quantization
+      range_b = tensor_range<float_t>(bias);
+      b_quantized =
+          float_tensor_to_quantized<uint8_t>(bias, range_b[0], range_b[1]);
+      quantized_add<uint8_t, uint8_t, int32_t>(a_requantized,
+        range_output_requantized[0], range_output_requantized[1],
+        b_quantized, range_b[0], range_b[1],
+        &a_quantized, &(range_output[0]), &(range_output[1]));
+
+      quantize_down_and_shrink_range<int32_t, uint8_t>(a_quantized,
+        range_output[0], range_output[1],
+        &(range_output_requantized[0]), &(range_output_requantized[1]),
+        &a_requantized);
+    }
+
+    // Transforming dequantize data in uint8_t to flaot,
+    // this could be removed within concatenated quantized network
+    a = quantized_tensor_to_float<uint8_t>(a_requantized,
```

Comment:
@bhack @edgarriba The range for each output tensors should also be stored in tensor to pass to the next layer in tiny-dnn, so do you have some good idea for this data passing strategy on codes? I think it's different from full precision kernels cause new tensors of ranges are introduced there as output, so this line of code is not necessary in an efficient quantization structure.
I want to only use a_requantized and range_output_requantized as output tensors.